### PR TITLE
ci: update tools

### PR DESCRIPTION
**Pull request by kentakozuka**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -8,18 +8,12 @@ linters-settings:
...
     - gosimple
     - govet
     - ineffassign
-    # Since we got the following error, staticcheck is disabled. After solving it, we'll enable it.
-    # pkg/account/apikeycacher/apikeycacher.go:284:12: SA1019: ptypes.UnmarshalAny is deprecated: Call the any.UnmarshalTo method instead. (staticcheck)
-    # - staticcheck
-    - structcheck
     - typecheck
     - unused
-    - varcheck
```
Comment (kentakozuka) [inline .golangci.yml:22]: Removed deprecated linters

File: .github/workflows/pr-go.yaml [hunk 2]
```diff
@@ -152,6 +160,6 @@ jobs:
           go-version: ${{ env.GO_VERSION }}
           cache: true
       - name: Install golangci-lint
-        run: go install github.com/golangci/golangci-lint/cmd/golangci-lint@${{ env.GOLANGCI_LINT }}
+        run: go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
```
Comment (cre8ivejp) [inline .github/workflows/pr-go.yaml:163]: Is there a reason to use the latest version?
This could introduce instabilities to the CI.

Comment (kentakozuka) [reply .github/workflows/pr-go.yaml:163]: I set latest here as goimport.
https://github.com/bucketeer-io/bucketeer/blob/c2a78137cf9bdbcd8dd5b01442b7e21b2a622d39/.github/workflows/pr-go.yaml#L65

Also, golangci-lint does not affect codes while mockgen and protoc do.

Comment (cre8ivejp) [reply .github/workflows/pr-go.yaml:163]: I asked because last year, an older version broke the CI due to new linters, deprecation, etc.

**Review by cre8ivejp — APPROVED** [review]
👍

**Pull request merged**